misc/proper_plugin.py

```diff
@@ -40,21 +42,40 @@ def isinstance_proper_hook(ctx: FunctionContext) -> Type:
     for arg in ctx.arg_types[0]:
         if is_improper_type(arg):
             right = get_proper_type(ctx.arg_types[1][0])
-            if isinstance(right, CallableType) and right.is_type_obj():
-                if right.type_object().fullname() in ('mypy.types.Type',
-                                                      'mypy.types.ProperType',
-                                                      'mypy.types.TypeAliasType'):
-                    # Special case: things like assert isinstance(typ, ProperType) are always OK.
-                    return ctx.default_return_type
-                if right.type_object().fullname() in ('mypy.types.UnboundType',
-                                                      'mypy.types.TypeVarType'):
-                    # Special case: these are not valid targets for a type alias and thus safe.
-                    return ctx.default_return_type
+            if is_special_target(right):
+                return ctx.default_return_type
```

Comment:
Idea: we could also maybe add a check for when the left operand is Any and the right operand is some Type. Could be overkill/out-of-scope for this PR.